Source/WebKit/UIProcess/WebProcessPool.cpp

```diff
@@ -1006,12 +1013,15 @@ void WebProcessPool::disconnectProcess(WebProcessProxy& process)
 
     // FIXME (Multi-WebProcess): <rdar://problem/12239765> Some of the invalidation calls of the other supplements are still necessary in multi-process mode, but they should only affect data structures pertaining to the process being disconnected.
     // Clearing everything causes assertion failures, so it's less trouble to skip that for now.
-    Ref<WebProcessProxy> protectedProcess(process);
+    Ref protectedProcess { process };
 
     m_backForwardCache->removeEntriesForProcess(process);
 
-    if (process.isRunningWorkers())
-        removeFromRemoteWorkerProcesses(process);
+    if (process.isRunningWorkers()) {
+        process.disableRemoteWorkers(RemoteWorkerType::ServiceWorker);
```

Comment:
These two functions are called together twice.  Maybe make it take an OptionSet?

Reply:
Good idea. Will do.